src/ui/ResultFolding/ResultFolding.ts

```diff
@@ -173,42 +163,37 @@ export class ResultFolding extends Component {
     this.results.appendChild(this.waitAnimation);
     this.updateElementVisibility();
 
-    let ret = this.moreResultsPromise.then((results?: IQueryResult[]) => {
-      this.childResults = results;
-      this.showingMoreResults = true;
-      this.usageAnalytics.logClickEvent<IAnalyticsDocumentViewMeta>(
-        analyticsActionCauseList.foldingShowMore,
-        this.getAnalyticsMetadata(),
-        this.result,
-        this.element
-      );
-      return this.displayThoseResults(results).then(() => {
-        this.updateElementVisibility(results.length);
-        return results;
-      });
-    });
+    const results: IQueryResult[] = await this.moreResultsPromise;
```

Comment:
Is the type needed here ? ( `IQueryResult[]` ). 
Does the compiler not detect it correctly otherwise ?